### DISCUSSION TITLE
Explanation of formatter defaults

### DISCUSSION THREAD

**Discussion by jsejcksn** [biomejs/biome#1382] [Q&A]
Hello: I created this because I am seeking knowledge that I didn't find by searching. This isn't an issue, so I'm not creating it as one, however — and I want to emphasize this part strongly — my goal is **not to create a discussion** about opinions — rather I simply want to understand this:

**For each default formatter value: what was the rationale behind choosing it?**

The [formatter documentation](https://biomejs.dev/formatter/) ([📌](https://github.com/biomejs/biome/tree/main/website/src/content/docs/formatter)) seems to suggest that it relies on the defaults of Prettier…

> Biome is an opinionated formatter that has the goal to stop all ongoing debates over styles. It follows a similar [philosophy to Prettier](https://prettier.io/docs/en/option-philosophy.html), only supporting a few options to avoid debates over styles, turning into debates over Biome options. It deliberately [resists the urge to add new options](https://github.com/prettier/prettier/issues/40) to prevent [bike-shed discussions](https://en.wikipedia.org/wiki/Law_of_triviality) in teams so they can focus on what really matters instead.

…and it lists some [Differences with Prettier](https://biomejs.dev/formatter/differences-with-prettier/) ([📌](https://github.com/biomejs/biome/blob/73c524454812c038c027b8e5442629d09ce6dfdd/website/src/content/docs/formatter/differences-with-prettier.md)), even explaining general [Formatter Option Philosophy](https://biomejs.dev/formatter/option-philosophy/) ([📌](https://github.com/biomejs/biome/blob/73c524454812c038c027b8e5442629d09ce6dfdd/website/src/content/docs/formatter/option-philosophy.md)), yet there are still other differences with Prettier's defaults that aren't included — one example is `tab` as the default [`indentStyle`](https://biomejs.dev/reference/configuration/#formatterindentstyle) ([📌](https://github.com/biomejs/biome/blob/73c524454812c038c027b8e5442629d09ce6dfdd/website/src/content/docs/reference/configuration.mdx#L421-L425)). (Perhaps that one was motivated by the justifications at this issue? [prettier/prettier#7475 — Change `useTabs` to `true` by default](https://github.com/prettier/prettier/issues/7475))

I found this issue, but it seems incomplete: [biomejs/biome#739 — ☂️ Prettier Divergence collection from the Prettier challenge](https://github.com/biomejs/biome/issues/739)

The [philosophy](https://biomejs.dev/internals/philosophy/) ([📌](https://github.com/biomejs/biome/blob/73c524454812c038c027b8e5442629d09ce6dfdd/website/src/content/docs/internals/philosophy.mdx)) includes this item in the "Project Management" section:

> **Clear messaging of decisions.** The team might evaluate options and make decisions using private channels. While the team will try to keep discussions using public channels like [GitHub discussions](https://github.com/biomejs/biome/discussions) or [Discord](https://discord.gg/BypW39g6Yc), frequent private check-in are the norm, due to the nature of the private company. When decisions occur via private channels, the team has to commit to communicate these decisions using the public channels.

I hope — toward that goal — a team member can share here the rationale for each formatter default (or a permalink if I've somehow missed it elsewhere). Thanks!

**Comment by zorro901** [comment]
rome had this discussion.
https://github.com/rome/tools/issues/425
